frontend/apps/reader/modules/readerlink.lua

```diff
@@ -678,9 +678,15 @@ function ReaderLink:onTap(_, ges)
     end
 end
 
---- Remember current location so we can go back to it
---- increments location_stack_index to point the the last item,
---- and clears the next item in case there was a Forward history
+-- Remember current location so we can go back to it
+-- increments location_stack_index to point the the last item,
+-- and clears the next item in case there was a Forward history.
+-- self.location_stack is not a stack, rather an array.
+-- We add locations to the end but can go backwards and fowards.
+-- Since this is an array all we need to make sure is that our last value is
+-- followed by a nil.git
+-- If we add a new element we nil index+1 to delete any old forward locations
+-- we cannot delete them earlier as we still need them when judt going back.
```

Comment:
_Judt_ is going back because he is followed by the infamous _nil.git_ ?! Scary story !